### ISSUE TITLE
Aliases for Tabs, Resizing controls

### ISSUE THREAD

**Issue by ghost** [vladimiry/ElectronMail#175]
I would like to suggest two cosmetic/convenience enhancements for the future development (maybe already in the pipeline, however I haven't found any related "issue" when doing a preliminary search):

1. Aliases for Tabs - simply said, naming e-mail tabs, e.g. Work, Personal, Shopping, Verification,...I hope it is not just my case, when these aliases could result in noticeably shorter tab labels, which is a great segue to the second one...

2. Resizing controls (in this case, I reckon, width-only resizing) for things like column with e-mail tabs when not using tab view mode (and account/e-mail names are longer) w/o word wrapping (or maybe with an option to toggle between both), or when using the local storage mode and to be able to drag-to-adjust width of e-mail list and preview window.

And maybe a one more, a tiny one, which I am sure is not in the pipeline and far from being a priority - dark mode, anyone?

**Comment by vladimiry** [comment]
Thanks for the suggestion. All of these points except the dark mode was considered initially when the related implementation was taking place but was not implemented nor put into the pipeline for different reasons (focus on another more critical things, limited capacity, no demonstrated users demand, etc).

For point one, we could get an additional view mode where the controls at the left side as shown [here](https://github.com/vladimiry/ElectronMail/blob/master/images/view-full.png) would be collapsed to the narrow column with the fixed width buttons and the stuff like the unread counter, view mode switch button, lock icon indicator, etc would be placed vertically at the bottom side instead of the currently hardcoded horizontal placing. Then alternatively or in addition, the labeling could be added as suggested. Besides that, in addition to the labeling, the coloring could be added so it would be easier to distinguish the email category based on the handle button color.

For point two it's now the 50%/50% hardcoded areas width splitting used for list/preview areas in database view mode but yes could be configurable using something like the slider handle placed in between the areas which a user could drag to change the areas width ratio.

Can't say anything specific about the dark mode at this time, have never explored this stuff.

And addition thing would be enabling hotkeys navigation in database view mode, at least the up/down arrow keys for navigating through the emails list.

Nevertheless, I'm not saying yet that the issue gets high priority.

**Comment by nil0x42** [comment]
Aliases would definitely be a great feature !

**Comment by vladimiry** [comment]
Adding aliases support is actually easy to do.

**Comment by vladimiry** [comment]
If someone wants to try the aliases feature the work-in-progress version is uploaded [here](https://www.dropbox.com/sh/il1jdf1n72vecpo/AAC-UltKdz7ZfzJieI_229X7a?dl=0&lst=).

I'm not releasing it yet as I want the new release to include some more changes like for example upgrading @electron v6=>v8 (currently blocked by some upstream bugs), dropping Tutanota support.

Few examples of using aliases thing (horizontal and vertical account handle buttons positioning):
![h-2](https://user-images.githubusercontent.com/1560781/69375978-8307d380-0cba-11ea-8d28-8c40289f5e58.jpg)

![v-2](https://user-images.githubusercontent.com/1560781/69375979-8307d380-0cba-11ea-9bc8-fec06eaf68ec.jpg)

Here is how to set the alias for the account (expand the "Extended Options" block):
![options-2](https://user-images.githubusercontent.com/1560781/69376049-a894dd00-0cba-11ea-8093-2bf7355d4e12.png)

Besides to reduce the handle buttons with even further the margins between the account handle icons got slightly decreased and the text on the options button removed (so only the icon remains).

**Comment by ylluminate** [comment]
@vladimiry this is actually working pretty nicely. There are a couple tweaks that are still needed to help this out: 

- please allow us to turn off the red background box count of messages in inbox. This would help further compression. 
- please make the "online/database" toggle (plug icon/database icon) smaller somehow.

It would be helpful to compress this further somehow, perhaps offering just an icon / brief (aliased) name + right click options to actually enact/enable certain functions/toggles instead of having them directly visible.

**Comment by ylluminate** [comment]
Another realization is that you could allow us to have a "zoom" or "size" control for the tab bar to simply shrink and enlarge it or return it to "normal" size permanently. That would allow you to keep everything the way you like it, but allow some of us to have a smaller management interface since right now it just seems to bulky / big.

**Comment by vladimiry** [comment]
The issue is not closed yet. So yes I agree there is a room for improvement.

**Comment by vladimiry** [comment]
> For point one, we could get an additional view mode where the controls at the left side as shown here would be collapsed to the narrow column with the fixed width buttons and the stuff like the unread counter, view mode switch button, lock icon indicator, etc would be placed vertically at the bottom side instead of the currently hardcoded horizontal placing.

![controls view modes (top, left, left-thing)](https://user-images.githubusercontent.com/1560781/74595784-e81f5b00-5056-11ea-80bd-a169371edad7.gif)

This is a new `left-thin` account handle buttons layout mode. Don't forget the aliases supported is enabled since [v4.0.0](https://github.com/vladimiry/ElectronMail/releases/tag/v4.0.0), so you can set for example a one-letter alias for the account.

**Comment by vladimiry** [comment]
Closing the issue as resolved.

The following point gets `won't fix` status for now:

> when using the local storage mode and to be able to drag-to-adjust width of e-mail list and preview window

**Comment by vladimiry** [comment]
If someone wants to try the not-yet-released build with `left-thin` account handle buttons layout mode option, it's uploaded [here](https://mega.nz/#F!h0IzhKRT!alKQxnPFdw_krZcin4pFmg) (download link picked from #253).

**Comment by ylluminate** [comment]
This is very interesting in terms of the new thin left bar view. I think I like it. I would suggest that the horizontal width is too much yet still as there's a lot of additional padding here on the left and right side of the text...

**Comment by ylluminate** [comment]
Ah, I see the problem, if I have too many unread messages on an account it will expand it out. It might be good to simply disable the inbox count and instead have a red dot for mailboxes that have new mail vs having an actual count. Anymore it's not worth having a count, but simply having an indicator that there are new messages and a simple dot would suffice for this.

**Comment by vladimiry** [comment]
> Ah, I see the problem, if I have too many unread messages on an account it will expand it out.

The edge case I missed to handle, so I will turn the unread number to vertical mode.

**Comment by ylluminate** [comment]
Hmm, I guess you are right that that is an edge case, but I still would prefer an option to turn off the unread message count and simply have a dot nonetheless. Message count does nothing for me and for many I know since we really simply need a "new message" visual indicator and nothing more usually.

**Comment by vladimiry** [comment]
> an option to turn off the unread message count

v4.4.0 will enable this feature.

![peek-2-min](https://user-images.githubusercontent.com/1560781/75092773-5484e780-558c-11ea-87dc-d22f0a248d65.gif)
